packages/storage/src/storage.ts

```diff
@@ -321,61 +323,59 @@ export class Storage {
         this.getGaiaAddress(opt.app!, opt.username, opt.zoneFileLookupURL),
       ]);
 
-      if (!fileContents) {
-        return fileContents;
-      }
-      if (!gaiaAddress) {
+      if (!fileContents) return fileContents;
+
+      if (!gaiaAddress)
         throw new SignatureVerificationError(
           `Failed to get gaia address for verification of: ${path}`
         );
-      }
-      if (!signatureContents || typeof signatureContents !== 'string') {
+
+      if (!signatureContents || typeof signatureContents !== 'string')
         throw new SignatureVerificationError(
           'Failed to obtain signature for file: ' +
             `${path} -- looked in ${path}${SIGNATURE_FILE_SUFFIX}`
         );
-      }
+
       let signature;
       let publicKey;
       try {
         const sigObject = JSON.parse(signatureContents);
         signature = sigObject.signature;
         publicKey = sigObject.publicKey;
       } catch (err) {
-        if (err instanceof SyntaxError) {
+        if (err instanceof SyntaxError)
           throw new Error(
-            'Failed to parse signature content JSON ' +
-              `(path: ${path}${SIGNATURE_FILE_SUFFIX})` +
-              ' The content may be corrupted.'
+            `Failed to parse signature content JSON (path: ${path}${SIGNATURE_FILE_SUFFIX}) The content may be corrupted.`
           );
-        } else {
-          throw err;
-        }
+        throw err;
       }
       const signerAddress = publicKeyToAddress(publicKey);
-      if (gaiaAddress !== signerAddress) {
+      if (gaiaAddress !== signerAddress)
         throw new SignatureVerificationError(
-          `Signer pubkey address (${signerAddress}) doesn't` +
-            ` match gaia address (${gaiaAddress})`
+          `Signer pubkey address (${signerAddress}) doesn't match gaia address (${gaiaAddress})`
         );
-      } else if (!verifyECDSA(fileContents, publicKey, signature)) {
+
+      if (
+        !verifyECDSA(
+          typeof fileContents === 'string'
+            ? utf8ToBytes(fileContents)
+            : new Uint8Array(fileContents),
```

Comment:
What's the type of `fileContents` here? Could possibly avoid a large copy

Reply:
In this case it should create a view (see https://github.com/hirosystems/stacks.js/pull/1343#discussion_r964781901)